Sign-up screen UI

ranaprachi opened the pull request.

shresthagrawal commented:
Hello @ranaprachi good work!

Just few suggestions

* I get these warnings when I start the project :
`Warning: React does not recognize the `justifyContent` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `justifycontent` instead. If you accidentally passed it from a parent component, remove it from the DOM element.`
`Warning: React does not recognize the `alignItems` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `alignitems` instead. If you accidentally passed it from a parent component, remove it from the DOM element.`

* Can you rebase the code code with current master.
* Was not making the fields corner as rounded as in design, an intentional change?

ranaprachi commented:
Hello @shresthagrawal, @falafeljan , I have fixed the warning that were coming earlier and added theme configurations to make design fixes as well, please review.

falafeljan commented on renderer/components/FlexContainer.js:
```diff
@@ -0,0 +1,16 @@
+            padding: 0;
+        `
+    }  
+`;
```
good approach, but styled-components is a bit tricky on that one. the props lambda function works for simple values of CSS properties like `flex-flow: ${props => props.flow || 'column'};`, but you will miss the interpolation of styled-components if you return the whole CSS.

more clean would be using the props callback for each and every value, or using the `css` helper:
```js
import styled, {css} from 'styled-components';

const FlexContainer = styled.div`
  ${props => css`
    display: flex;
    ...
  `};
`;
```

falafeljan commented on renderer/components/Input.js:
```diff
@@ -3,7 +3,10 @@ import { Input as BootstrapInput} from '@bootstrap-styled/v4';
+      height: 45px;
+      box-sizing: border-box;
+      padding: 20px;
+      border-radius: 8px;
```
same as above. here, you even don't need the props lambda. simply do it like
```js
const Input = stlyed(...)`
  height: 34px;
  ...
`;
```

falafeljan commented on renderer/components/Page.js:
```diff
@@ -13,6 +13,22 @@ const GlobalStyle = createGlobalStyle`
+    font-family: 'Bowlby One';
+    font-style: normal;
+    font-weight: 400;
+    src: local('Bowlby One Regular'), local('BowlbyOne-Regular'), url(https://fonts.gstatic.com/s/bowlbyone/v11/taiPGmVuC4y96PFeqp8sqomI_L8zYK8.woff2) format('woff2');
```
please add the webfonts so that the application works locally.

ranaprachi replied:
Hi @falafeljan, yes I was planning to resolve this comment in later commits. Need to figure out how to include the right webpack loader for the font files.

falafeljan replied:
cool, thanks. good point, you could simply try this approach (using `file-loader`):
https://www.chriscourses.com/blog/loading-fonts-webpack

```js
module.exports = {
  ...
  module: {
    loaders: [
      ...,
      {
        test: /\.(woff(2)?|ttf|eot|svg)(\?v=\d+\.\d+\.\d+)?$/,
        use: [
          {
            loader: 'file-loader',
            options: {
              name: '[name].[ext]'
            }
          }
        ]
      }
    ]
  },
  ...
}
```

falafeljan commented on renderer/components/Page.js:
```diff
@@ -13,6 +13,22 @@ const GlobalStyle = createGlobalStyle`
+    font-family: 'Roboto';
+    font-style: normal;
+    font-weight: 400;
+    src: local('Roboto'), local('Roboto-Regular'), url(https://fonts.gstatic.com/s/roboto/v20/KFOmCnqEu92Fr1Mu4mxKKTU1Kg.woff2) format('woff2');
```
same as above.

falafeljan commented on renderer/components/Size.js:
```diff
@@ -0,0 +1,20 @@
+            min-height: ${props.minHeight || ''};
+            height: ${props.height || ''};
+            max-height: ${props.maxHeight || ''};
+        `
```
same as above in regard to `props` and the `css` helper.

falafeljan commented on renderer/components/Space.js:
```diff
@@ -0,0 +1,13 @@
+        props => `
+            padding: ${props.padding || ''};
+            margin: ${props.margin || ''};
+        `
```
same as above in regard to `props` and the `css` helper.

falafeljan commented on renderer/components/Text.js:
```diff
@@ -0,0 +1,14 @@
+        font-family: ${props.family || "Roboto"};
+        line-height: ${props.lineHeight || '1'};
+        text-align: ${props.align || "left"};
+   `}
```
same as above in regard to `props` and the `css` helper.

nice take that you're using the theme here. yet, i don't know whether we need to make all CSS properties props-equipped here. you could simply wrap the `Text` component via `styled(Text)` if you need to extend it or change properties.

ranaprachi replied:
I was planning to design it in such a way that it could be reused directly where-ever required and only extend it via styled(Text) in places where we need to implement something different then common use-cases.

falafeljan commented on renderer/pages/home.js:
```diff
@@ -9,21 +9,13 @@ import useProjects from '../effects/useProjects';
 import useUser from '../effects/useUser';
 import Page from '../components/Page';
 import Header from '../components/Header';
+import FlexContainer from '../components/FlexContainer';
```
nice!

falafeljan commented:
good stuff, thanks for addressing this. shall we merge now? @shresthagrawal @ranaprachi
